api/src/services/comments.ts

```diff
@@ -0,0 +1,125 @@
+import { useEnv } from '@directus/env';
+import { ErrorCode, InvalidPayloadError, isDirectusError } from '@directus/errors';
+import type { Accountability, Comment, PrimaryKey } from '@directus/types';
+import { uniq } from 'lodash-es';
+import { useLogger } from '../logger/index.js';
+import type { AbstractServiceOptions, MutationOptions } from '../types/index.js';
+import { getPermissions } from '../utils/get-permissions.js';
+import { isValidUuid } from '../utils/is-valid-uuid.js';
+import { Url } from '../utils/url.js';
+import { userName } from '../utils/user-name.js';
+import { AuthorizationService } from './authorization.js';
+import { ItemsService } from './items.js';
+import { NotificationsService } from './notifications.js';
+import { UsersService } from './users.js';
+
+const env = useEnv();
+const logger = useLogger();
+
+export class CommentsService extends ItemsService {
+	notificationsService: NotificationsService;
+	usersService: UsersService;
+
+	constructor(options: AbstractServiceOptions) {
+		super('directus_comments', options);
+		this.notificationsService = new NotificationsService({ schema: this.schema });
+		this.usersService = new UsersService({ schema: this.schema });
+	}
+
+	override async createOne(data: Partial<Comment>, opts?: MutationOptions): Promise<PrimaryKey> {
+		if (!data['collection'] || !data['item'] || !data['comment']) {
+			throw new InvalidPayloadError({ reason: 'Missing required fields' });
+		}
+
+		const authorizationService = new AuthorizationService({ schema: this.schema, accountability: this.accountability });
+
+		await authorizationService.checkAccess('read', data['collection'], data['item']);
```

Comment:
We should probably have this check for update as well, it would suffer from the same issues as create if left without it

Reply:
I think the item access check should only be for create. For example, there could be a comments moderator who should be able to update or delete comments without needing to access the item.

Reply:
Updated the recommended permission to only allow for the updating of comments too. https://github.com/directus/directus/pull/22295/commits/0cd367d4f09df15c588b5f37fb578a23916847e0